### ISSUE TITLE
Hyper-V MINIX: No network hardware interfaces detected!

### ISSUE THREAD

**Issue by mcandre** [hashicorp/packer#5605]
FOR BUGS:

The Hyper-V provider does not appear to provide a virtual network interface supported by MINIX guests.

Is there a way for packer JSON to configure Hyper-V to specify a different virtual network interface to the guest, perhaps a more generic interface that legacy guests integrate with more easily?

# System

- Packer version from `packer version`: `v1.1.3-dev`
- Host platform: Windows 10 Pro

# Configuration

minix.json:

```json
{
  "builders": [
    {
      "type": "hyperv-iso",
      "vm_name": "minix",
      "iso_url": "minix_R3.4.0rc6-d5e4fc0.iso",
      "iso_checksum_type": "md5",
      "iso_checksum": "55f8ca78ee6bdb7c12c02c69a2925439",
      "ssh_username": "vagrant",
      "ssh_password": "vagrant",
      "ssh_timeout": "1800s",
      "disk_size": 1000,
      "ram_size": 1024,
      "boot_command": [
        "<enter><wait10><wait10><wait10><wait10><wait10><wait10><wait10><wait10>",
        "root<enter><wait>",
        "setup<enter><enter><wait>",
        "<enter><enter><wait10><wait10>",
        "<enter><wait>",
        "<enter><wait5>",
        "yes<enter><wait10><wait10><wait10>",
        "<enter><wait>",
        "<enter><wait>",
        "<enter><wait10><wait10><wait10><wait10><wait10><wait10><wait10><wait10><wait10><wait10><wait10><wait10><wait10><wait10><wait10><wait10><wait10><wait10>",
        "<enter><wait>",
        "<enter><wait>",
        "<enter><wait>"
      ],
      "shutdown_command": "su root -c '/sbin/halt -p'"
    }
  ],
  "provisioners": [
    {
      "type": "shell",
      "execute_command": "{{.Vars}} su -m root -c 'sh \"{{.Path}}\"'",
      "scripts": [
        "boot-sshd.minix.sh"
      ]
    }
  ],
  "post-processors": [
    {
      "type": "vagrant",
      "output": "minix-{{.Provider}}.box",
      "compression_level": "9",
      "vagrantfile_template": "Vagrantfile"
    }
  ]
}
```

The rest of the `boot_command` steps are elided in order to highlight the network setup error. At this point, MINIX installation exits unsuccessfully, and even fails to properly write the installation to disk. If the guest is sent the "reboot<enter>" keyboard characters, the resulting boot does not actually enter the installed MINIX system, but instead boots again into a fresh live CD MINIX system, causing the `user add -m -G wheel vagrant<enter>` section to fail, as the live CD does not provide user/group modification features. The rest of the `boot_command`, which does work with the VirtualBox provider, is as follows:

```json
"reboot<enter><wait10>",
"<enter><wait10><wait10><wait10><wait10><wait10>",
"root<enter><wait>",
"user add -m -G wheel vagrant<enter><wait>",
"passwd vagrant<enter><wait>",
"vagrant<enter><wait>",
"vagrant<enter><wait>",
"vagrant<enter><wait>",
"mkdir -p /home/vagrant/.ssh<enter><wait>",
"echo 'ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEA6NF8iallvQVp22WDkTkyrtvp9eWW6A8YVr+kz4TjGYe7gHzIw+niNltGEFHzD8+v1I2YJ6oXevct1YeS0o9HZyN1Q9qgCgzUFtdOKLv6IedplqoPkcmF0aYet2PkEDo3MlTBckFXPITAMzF8dJSIFo9D8HfdOV0IAdx4O7PtixWKn5y2hMNG0zQPyUecp4pzC6kivAIhyfHilFR61RGL+GPXQ2MWZWFYbAGjyiYJnAmCP3NOTd0jMZEnDkbUvxhMmBYSdETk1rRgm+R4LOzFUGaHqHDLKLX+FIPKcF96hrucXzcWyLbIbEgE98OHlnVYCzRdK8jlqm8tehUc9c9WhQ== vagrant insecure public key' >/home/vagrant/.ssh/authorized_keys<enter><wait>",
"chown -R vagrant: /home/vagrant/.ssh<enter><wait>",
"chmod 0700 /home/vagrant/.ssh<enter><wait>",
"chmod 0600 /home/vagrant/.ssh/authorized_keys<enter><wait>",
"yes | pkgin update<enter><wait10>",
"yes | pkgin install openssh<enter><wait10>",
"sh /usr/pkg/etc/rc.d/sshd onestart<enter><wait>"
```

boot-sshd.minix.sh:

```sh
#!/bin/sh

echo 'sshd=YES' >>/etc/rc.conf &&
    mkdir -p /etc/rc.d &&
    cp /usr/pkg/etc/rc.d/sshd /etc/rc.d/
```

Vagrantfile:

```ruby
Vagrant.configure("2") do |config|
  config.vm.guest = :netbsd
  config.ssh.shell = "sh"
  config.ssh.sudo_command = "su -m root -c '%c'"
  config.vm.synced_folder '.', '/vagrant', disabled: true
end
```

**Comment by kikitux** [comment]
As today the network seems not to be configurable.

so 2 options.

1. you can create one by hand, and then use hyper vmcx to use this vm
2. check what can be changed, potentially PR.

the code that creates the VM is here:
https://github.com/hashicorp/packer/blob/a51885be9c6e77aa6cd0c07114b2631af5ef76df/common/powershell/hyperv/hyperv.go#L190-L242

**Comment by tuxillo** [comment]
Hi,

Maybe this helps?
https://www.altaro.com/hyper-v/work-hyper-v-virtual-network-adapters/

Regards,
Antonio Huete

**Comment by SwampDragons** [comment]
Looks like this was addressed in #7128, so I'm closing.
